hawk/src/main/java/com/orhanobut/hawk/Hawk.java

```diff
@@ -111,20 +102,90 @@ public static void init(Context context, String password, LogLevel logLevel) {
      * @return true if put is successful
      */
     public static <T> boolean put(String key, List<T> list) {
+        return put(key, null, list, true, true) != null;
+    }
+
+    /*
+     * A helper put method.
+     *
+     * @param key          the key
+     * @param value        the value if isList is false, otherwise null
+     * @param list         the list, if isList is true, otherwise null
+     * @param isList       determines if a list instance should be present
+     * @param addToStorage determines if the keyval should be immediately added to storage
+     */
+    private static <T> String put(String key, T value, List<T> list, boolean isList,
```

Comment:
for the preparing data, we should use a different name not "put". This method signature makes confusing.